packages/elements/src/button/button.styles.js

```diff
@@ -0,0 +1,252 @@
+import { colors, theme, spacers } from '@dhis2/ui-constants'
+import css from 'styled-jsx/css'
+
+export default css`
+    button {
```

Comment:
The button styles are maybe not as indicative of the rest of our lib, but I do think they demonstrate a potential issue. When you lint these styles with a css linter, a couple issues pop up, like the descending specificity in the selectors (which you don't want), and other issues. Overall I also think the styles are too complex and elaborate for a style primitive. Two things come to mind:

- Since elements are going to be mainly style building blocks, and those styles are going to be expressed in css, I think we should enable linting for the css used in the elements. That seems like a critical first line of defense.
- It would be good if we could somehow limit the complexity of the styles for style primitives. I think that's a potential pitfall. We're already limiting the complexity for elements by only allowing styles, but I can see things going off the rails with numerous overly complex and hard to debug styles as well. I can imagine there's going to be some pressure on elements to become too complex stylewise, and I think it'd be good to have something in place that'll help us resist that pressure.

Reply:
I agree that these styles are pretty heavy for a button. Linting would be a nice heads-up that we are doing something strange in CSS just as it is otherwise.